docs/analyzer-rules/AZFW0015.md

```diff
@@ -0,0 +1,44 @@
+# AZFW0015: Missing HttpResult attribute for multi-output function
+
+| | Value |
+|-|-|
+| **Rule ID** |AZFW00015|
+| **Category** |[Usage]|
+| **Severity** |Error|
+
+## Cause
+
+This rule is triggered when a multi-output function is missing a HttpResultAttribute on the HTTP response type.
+
+## Rule description
+
+For [functions with multiple output bindings](https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide?tabs=windows#multiple-output-bindings) using ASP.NET Core integration, the property correlating with the HTTP response needs to be decorated with the `HttpResultAttribute` in order to write the HTTP response correctly. This does not apply to properties of the type `HttpResponseData`.
+
+## How to fix violations
+
+Add the attribute `[HttpResult]` (or `[HttpResultAttribute]`) to the relevant property. Example:
+
+```csharp
+public static class MultiOutput
+{
+    [Function(nameof(MultiOutput))]
+    public static MyOutputType Run([HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequestData req,
+        FunctionContext context)
+    {
+        ...
+    }
+}
+
+public class MyOutputType
+{
+    [QueueOutput("myQueue")]
+    public string Name { get; set; }
+
+    [HttpResult]
+    public IActionResult HttpResponse { get; set; }
+}
+```
+
+## When to suppress warnings
+
+This rule should not be suppressed because this error will prevent the HTTP response from being written correctly.
```

Comment:
Would this lead to a failure? If so, we should state that here

Reply:
It wouldn't lead to a hard failure, but the customer code will not work as expected (issue https://github.com/Azure/azure-functions-dotnet-worker/issues/2079).